stats.js

```diff
@@ -2,21 +2,87 @@ const AQ = require('./gofish/aquarium.js');
 const connect = require("./connect.js");
 
 async function stats() {
+    let operation_types = await get_operation_types();
 
-    console.log("Retrieving operation type stats");
-
-    await connect();
-    let operation_types = await AQ.OperationType.all();
-    
-    for ( let i=0; i<operation_types.length; i++ ) {
+    for (let i = 0; i < operation_types.length; i++) {
         let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
         let stats = JSON.parse(response.data);
         console.log(`${i}: ${operation_types[i].category}/${operation_types[i].name}`);
-        for(const [key,val] of Object.entries(stats)) {
+        for (const [key, val] of Object.entries(stats)) {
             console.log("\t" + key + ": " + val);
         }
     }
+}
+
+async function most_recent() {
+    let operation_types = await get_operation_types();
+    let stat_map = {};
+
+    for (let i = 0; i < operation_types.length; i++) {
+        let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
+        let stats = JSON.parse(response.data);
+        console.log(`Processing ${operation_types[i].category}/${operation_types[i].name}`);
+        if (stats.last_run) {
+            stat_map[i] = stats;
+        }
+    }
+
+    let sorted_stats = Object.keys(stat_map).map(function(key) {
+        return [key, stat_map[key]];
+    });
+
+    sorted_stats.sort(function (fst, snd) {
+        return snd[1].last_run.localeCompare(fst[1].last_run);
+    });
+
+    console.log("Retrieving top 10 most recent runs");
 
+    print_top_10(operation_types, sorted_stats);
+}
+
+async function most_done() {
+    let operation_types = await get_operation_types();
+    let stat_map = {};
+
+    for (let i = 0; i < operation_types.length; i++) {
+        let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
+        let stats = JSON.parse(response.data);
+        console.log(`Processing ${operation_types[i].category}/${operation_types[i].name}`);
```

Comment:
If you use something like
process.stdout.clearLine(); 
process.stdout.cursorTo(0); 
process.stdout.write(Math.floor(100*i/operation_types.length) + "%");
you could get a progress percentage. You could put this in a separate method call
show_progress(i,n)

Reply:
done!